### PR TITLE
WRO-7525: Add search function 

### PR DESCRIPTION
**Pull request by sjro**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
After applying WRO-6288, the existing search function does not work

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Use @gatsby-contrib/gatsby-plugin-elasticlunr-search plugin (https://www.gatsbyjs.com/plugins/@gatsby-contrib/gatsby-plugin-elasticlunr-search) for supporting site search.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-6288, WRO-7524, WRO-7525

### Comments
Enact-DCO-1.0-Signed-off-by: SJ RO (sj.ro@lge.com)

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -2,6 +2,18 @@
 /* eslint-disable camelcase */
 const path = require('path');
 
+let compareslug = '';
```
Comment (juwonjeong) [inline gatsby-config.js:5]:
```suggestion
let moduleSlug= '';
```
I'm not sure about the suggestion but could you please rename this variable?

File: gatsby-config.js [hunk 2]
```diff
@@ -80,6 +92,80 @@ module.exports = {
...
+						}
+					},
+					title: (node) => {
+						if (isNotEmpty(node.frontmatter.title)) {
+							return node.frontmatter.title;
+						} else {
+							if (isNotEmpty(moduleName)) {
+								return moduleName;
+							}
+						}
```
Comment (juwonjeong) [inline gatsby-config.js:132]:
```suggestion
						if (isNotEmpty(node.frontmatter.title) && isNotEmpty(moduleName)) {
							return node.frontmatter.title;
						} 
```
I think this if-else statement could be made simpler.

File: gatsby-config.js [hunk 3]
```diff
@@ -80,6 +92,80 @@ module.exports = {
...
+					},
+					description: (node) => {
+						if (isNotEmpty(node.frontmatter.title)) {
+							return node.internal.content;
+						} else {
+							if (isNotEmpty(moduleName)) {
+								return node.internal.content;
+							}
+						}
```
Comment (juwonjeong) [inline gatsby-config.js:141]: I think this if-else statement could be made simpler.
```suggestion
						if (isNotEmpty(node.frontmatter.title) && isNotEmpty(moduleName)) {
							return node.internal.content;
						} 
```

File: src/components/SiteHeader/SiteHeader.js [hunk 4]
```diff
@@ -50,7 +67,7 @@ const SiteHeaderBase = kind({
 						</Cell>
 						<Cell>
 							<div className={css.siteSearch}>
-								<Search location={location} />
+								<SearchQueryItem location={location}/>
```
Comment (juwonjeong) [inline src/components/SiteHeader/SiteHeader.js:70]:
```suggestion
								<SearchQueryItem location={location} />
```

File: gatsby-config.js [hunk 5]
```diff
@@ -80,6 +92,80 @@ module.exports = {
...
+								if (isNotEmpty(jsnode.name) && jsnode.fields.slug.includes(compareslug)) {
+									if (isNotEmpty(jsnode.description___NODE) && !jsnode.name.includes('/')) {
+										const descNode = getNode(jsnode.description___NODE);
+										memberContent += jsnode.name + " ";
+										memberDescriptionContent += descNode.internal.content + " ";
+									}
+								}
+							});
+							return memberDescriptionContent;
+						}
```
Comment (juwonjeong) [inline gatsby-config.js:158]: I think duplicated words are combined to memberContent.
For example, core/keymap module, "members" field includes those words.
"keyCode removeAll set keyCode addAll add keyCode name name name is set remove "
Some words are duplicated in members field.
Would it be better to remove this duplicate word or leave it as is?
```suggestion
						memberContent = "";
						const memberNameSet = new Set();
						let memberDescriptionContent = "";
						if (!isNotEmpty(node.frontmatter.title) && isNotEmpty(moduleName)) {
							const documentationJSNodes = getNodesByType(`DocumentationJs`);
							documentationJSNodes.forEach((jsnode) => {
								if (isNotEmpty(jsnode.name) && jsnode.fields.slug.includes(compareslug)) {
									if (isNotEmpty(jsnode.description___NODE) && !jsnode.name.includes('/')) {
										const descNode = getNode(jsnode.description___NODE);
										memberNameSet.add(jsnode.name);
										memberDescriptionContent += descNode.internal.content + " ";
									}
								}
							});
							
							memberNameSet.forEach((nodeName) => {
								memberContent += nodeName + " ";
							});
							return memberDescriptionContent;
						}
```

**Review by juwonjeong — COMMENTED** [review]
Resolve the conflict, and I think feature/WRO-6288 needs to be merged to this branch because it requires --no-cone(git sparse-checkout) option.

File: gatsby-config.js (hunk 5 above)
Comment (sjro) [reply gatsby-config.js]: It is better to leave it as it is now, as those words become the key for searching and can go to different page links with the same word.

File: gatsby-config.js (hunk 3 above)
Comment (sjro) [reply gatsby-config.js]: fixed it

File: gatsby-config.js (hunk 5 above)
Comment (juwonjeong) [reply gatsby-config.js]: Okay, I understand.

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**